### ISSUE TITLE
Possibly removed implicit `this` argument

### ISSUE THREAD

**Issue by giann** [buzz-language/buzz#98]
Since buzz is supposed to be _unambiguous_ do we need implicit `this` argument for objects' methods?

```buzz
object Person {
    str name,

    fun sayHello(str to) -> print("Hello {to} from {this.name}");
}
```

would become

```buzz
object Person {
    str name,

    fun sayHello(Person this, str to) -> print("Hello {to} from {this.name}");
}
```

**Comment by notcancername** [comment]
Wouldn't this incorrectly suggest a copy is being made?

**Comment by giann** [comment]
No, objects, lists and maps are always passed by reference and not copy.

**Comment by hexaredecimal** [comment]
Wouldn't it be better if the type of "this" can be omitted and inferred from the object block because the Object is a type... right...? In this case "Person" is?

**Comment by giann** [comment]
If we do this, it means we don't have "methods" anymore, just functions associated with the object. `this` becomes a user convention and buzz would see it like a normal function argument.

Also I try to avoid inferring when I can help it. One of buzz tag lines is that it's unambiguous, meaning very few implicit behaviors.
